tensorflow/tools/ci_build/builds/test_on_install.sh

```diff
@@ -0,0 +1,211 @@
+
+# If NO_TEST_ON_INSTALL is set to any non-empty value, skip all Python
+# tests-on-install and exit right away
+if [[ ! -z ${NO_TEST_ON_INSTALL} ]]; then
+  echo "NO_TEST_ON_INSTALL=${NO_TEST_ON_INSTALL}:"
+  echo "  Skipping ALL Python unit tests on install"
+  exit 0
+fi
+
+# Directory from which the unit-test files will be run
+PY_TEST_DIR="/tmp/tf_py_tests_${RAND_STR}"
+rm -rf ${PY_TEST_DIR}
+mkdir ${PY_TEST_DIR}
+
+# Copy source files that are required by the tests but are not included in the
+# PIP package
+LIB_PYTHON_DIR=`ls -d ${HOME}/.local/lib/python*`
+PACKAGES_DIR=`ls -d ${LIB_PYTHON_DIR}/*-packages`
+
+echo "Copying some source directories that are required by tests but are "\
+"not included in install to Python packages directory: ${PACKAGES_DIR}"
+
+# tensorflow.python.tools
+rm -rf ${PACKAGES_DIR}/tensorflow/python/tools
+cp -r /tensorflow/tensorflow/python/tools \
+      ${PACKAGES_DIR}/tensorflow/python/tools
+touch ${PACKAGES_DIR}/tensorflow/python/tools/__init__.py  # Make module visible
+
+echo "Copying additional files required by tests to working directory "\
+"for test: ${PY_TEST_DIR}"
+
+# Image files required by some tests, e.g., images_ops_test.py
+mkdir -p ${PY_TEST_DIR}/tensorflow/core/lib
+rm -rf ${PY_TEST_DIR}/tensorflow/core/lib/jpeg
+cp -r /tensorflow/tensorflow/core/lib/jpeg ${PY_TEST_DIR}/tensorflow/core/lib
+rm -rf ${PY_TEST_DIR}/tensorflow/core/lib/png
+cp -r /tensorflow/tensorflow/core/lib/png ${PY_TEST_DIR}/tensorflow/core/lib
+
+# Run tests
+DIR0=`pwd`
+ALL_PY_TESTS=`find tensorflow/python -name "*_test.py"`
+PY_TEST_COUNT=`echo ${ALL_PY_TESTS} | wc -w`
+
+if [[ ${PY_TEST_COUNT} -eq 0 ]]; then
+  echo "ERROR: Cannot find any tensorflow Python unit tests to run on install"
+  exit 1
+fi
+
+# Iterate through all the Python unit test files using the installation
+COUNTER=0
+PASS_COUNTER=0
+FAIL_COUNTER=0
+SKIP_COUNTER=0
+FAILED_TESTS=""
+
+for TEST_FILE_PATH in ${ALL_PY_TESTS}; do
+  ((COUNTER++))
+
+  # Copy to a separate directory to guard against the possibility of picking up 
+  # modules in the source directory 
+  cp ${TEST_FILE_PATH} ${PY_TEST_DIR}/
+
+  # If PY_TEST_WHITELIST is not empty, only the white-listed tests will be run
+  if [[ ! -z ${PY_TEST_WHITELIST} ]] && \
+     [[ ! ${PY_TEST_WHITELIST} == *"${TEST_FILE_PATH}"* ]]; then
+    ((SKIP_COUNTER++))
+    echo "Skipping non-whitelisted test: ${TEST_FILE_PATH}"
+    continue
+  fi
+
+  # If the test is in the black list, skip it
+  if [[ ${PY_TEST_BLACKLIST} == *"${TEST_FILE_PATH}"* ]]; then
+    ((SKIP_COUNTER++))
+    echo "Skipping blacklisted test: ${TEST_FILE_PATH}"
+    continue
+  fi
+
+  TEST_FILE_BASENAME=`basename "${TEST_FILE_PATH}"`
+
+  TEST_OUTPUT=${PY_TEST_DIR}/${TEST_FILE_BASENAME}.out
+
+  # Before running the test, cd away from the Tensorflw source to 
+  # avoid the possibility of picking up dependencies from the 
+  # source directory
+  cd ${PY_TEST_DIR} 
+  python ${PY_TEST_DIR}/${TEST_FILE_BASENAME}
```

Comment:
You can avaoid all the noise if you forward stderr and stdout to a file. Current successful build has 99x string "ERROR:" in it, and a lot of stacktraces. Is it not confusing? You redirect output to a file, cat that file on error - bellow "Python test-on-install FAILED". Also you can keep those files around in the workspace (some subdirectory) if somebody want to see them.